test/test_output_writer.cpp

```diff
@@ -2,29 +2,29 @@
 #include <GraphMol/SmilesParse/SmilesParse.h>
 
 #include <coaler/io/OutputWriter.hpp>
+#include <vector>
 
 #include "catch2/catch.hpp"
+using namespace coaler::multialign;
+using namespace coaler::io;
 
 TEST_CASE("Output Parser", "[io]") {
-    coaler::io::OutputWriter output_parser;
-
     SECTION("Add one pair to output") {
-        RDKit::RWMol *mol_a = RDKit::SmilesToMol("CCCN");
-        RDKit::RWMol *mol_b = RDKit::SmilesToMol("CCCO");
+        const auto mol_a = RDKit::SmilesToMol("CCCN");
+        const auto mol_b = RDKit::SmilesToMol("CCCO");
 
         RDKit::DGeomHelpers::EmbedMolecule(*mol_a);
         RDKit::DGeomHelpers::EmbedMolecule(*mol_b);
 
-        coaler::io::AlignedMolPair mol_pair;
-        mol_pair.mol_a = RDKit::ROMol{*mol_a};
-        mol_pair.mol_b = RDKit::ROMol{*mol_b};
-        mol_pair.id_mol_a = 0;
-        mol_pair.id_mol_b = 1;
-        mol_pair.align_score = 0.55;
+        const auto lig_a = Ligand(*mol_a, UniquePoseSet{0}, 0);
+        const auto lig_b = Ligand(*mol_b, UniquePoseSet{0}, 0);
+
+        auto score = 0.55;
+        const MultiAlignerResult result(score,
+                std::unordered_map<LigandID, PoseID>{{0, 0}, {1, 1}},
+                std::vector<Ligand>{lig_a, lig_b});
 
-        output_parser.add_aligned_mols(mol_pair);
 
-        SECTION("save output in file") { output_parser.save_molecules_w_scores_in_file("output"); };
-        SECTION("print output in log") { output_parser.print_in_log_molecules_w_scores(); };
+        SECTION("save output in file") { OutputWriter::save_molecules_w_scores_in_file("/tmp/", result); };
     };
```

Comment:
I'd prefer if we ensured that the correct moles are actually written to the file. e.g. read the file, check number and identity of molecues and maybe also the coordinates?